web3/gas_strategies/time_based.py

```diff
@@ -60,11 +61,15 @@ def _aggregate_miner_data(raw_data):
 
     for miner, miner_data in data_by_miner.items():
         _, block_hashes, gas_prices = map(set, zip(*miner_data))
+        try:
+            price_percentile = percentile(gas_prices, percentile=15)
+        except InsufficientData:
+            price_percentile = min(gas_prices)
```

Comment:
Just because `min()` is what was being calculated before the percentile function.  In a private network with just a few transactions per block, wouldn't using the last block `max(prices)` potentially send the price on a continual upward trend?

Reply:
I suppose, but it could also trend down with min. Also maybe it doesn't matter on a private network? I don't feel strongly about it.

Reply:
Couldn't you also include transactions from the last few blocks when there is not enough data available?

Reply:
It's already looking across 120 blocks by default, but this is grouped by miner. So a few scenarios could cause a small data set:

- we are near the beginning of the chain, with only a few blocks to sample
- even with many blocks and transactions, maybe this particular miner has only mined a few blocks/transactions